src/health.c

```diff
@@ -520,17 +523,60 @@ static inline RRDVAR *rrdvar_create_and_index(const char *scope, avl_tree_lock *
     else {
         // already exists
         freez(variable);
-        rv = NULL;
```

Comment:
this line is really needed
without it, netdata will double-free all the dimension variables that overlap somehow.